### ISSUE TITLE
False-positive "no-member" on subclassed dataclass

### ISSUE THREAD

**Issue by mkielar** [PyCQA/pylint#3754]
### Steps to reproduce
Self-contained example: [sscce.zip](https://github.com/PyCQA/pylint/files/4986912/sscce.zip)

1. Take this file:
    ```python
    from abc import ABCMeta, abstractmethod
    import dataclasses as dc
    from typing import Any, Dict

    @dc.dataclass(frozen=True)
    class DeploymentState(metaclass=ABCMeta):
        type: str

        @abstractmethod
        def to_dict(self) -> Dict:
            """
            Serializes given DeploymentState instance to Dict.
            :return:
            """

    @dc.dataclass(frozen=True)
    class DeploymentStateEcs(DeploymentState):
        blue: Any
        green: Any
        candidate: Any

        def to_dict(self) -> Dict:
            return {
                'type': self.type,
                'blue': dc.asdict(self.blue),
                'green': dc.asdict(self.green),
                'candidate': self.candidate.value,
            }

    @dc.dataclass(frozen=True)
    class DeploymentStateLambda(DeploymentState):
        current: Any
        candidate: Any

        def to_dict(self) -> Dict:
            return {
                'type': self.type,
                'current': dc.asdict(self.current),
                'candidate': dc.asdict(self.candidate) if self.candidate else None,
            }
    
    ```
2. `pylintrc` file:
    ```
    [MESSAGES CONTROL]
    disable=fixme,
        logging-too-many-args,
        logging-fstring-interpolation,
        missing-docstring,
        no-else-return,
        too-few-public-methods, 
    ```
3. Run:
   ```
   pylint test.py
   ```

### Current behavior
`pylint` complains about a missing field:
```
→ pylint test.py
************* Module test
test.py:24:20: E1101: Instance of 'DeploymentStateEcs' has no 'type' member (no-member)
test.py:37:20: E1101: Instance of 'DeploymentStateLambda' has no 'type' member (no-member)

------------------------------------------------------------------
Your code has been rated at 4.12/10 (previous run: 4.12/10, +0.00)
```

### Expected behavior
The field is obviously there, `pylint` should see it.

### pylint --version output
```
→ pylint --version
pylint 2.5.3
astroid 2.4.2
Python 3.8.3 (default, Feb 26 2020, 00:00:00)
[GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]
```

**Comment by david-yz-liu** [comment]
Cool, thanks for the prompt merge @Pierre-Sassoulas!
